### PR TITLE
Fill documentation usage

### LINKED ISSUE #1187: [Library]: Fill missing documentation's examples
**Description**

Look for functions that have its usage example not implemented and implement this.

A concrete function that was found in this state is `coalesce`.

**Related**

> Speaking of coalesce, there are no examples for it in the documentation.

[Krenium @ discord.com][message]

[message]: https://discord.com/channels/765519132186640445/829324913097048065/1105995742826475614

### PR DESCRIPTION
**Pull request by RickBarretto**
# Description

Add some missing usage examples to documentation.

## Tasks
- [x] prepend
- [x] infinite?
- [x] coalesce
- [x] range (❗)

## Related
- Fixes #1187
- Removes #1170's TODO

## Type of change
- [x] Add/Update documentation
- [x] This change requires a documentation update

### CODE REVIEW

File: src/library/Numbers.nim [hunk 1]
```diff
@@ -864,6 +864,15 @@ proc defineSymbols*() =
...
         example     = """
+            infinite? 4             ; false
+            infinite? infinite      ; true
+            infinite? ∞             ; true
+            ..........
+            a: infinite
+            infinite? a             ; true
+            
+            b: 0
+            infinite? b             ; true
```
Comment (drkameleon) [inline src/library/Numbers.nim:875]:
```suggestion
            infinite? b             ; false
```

I think this is a tiny typo... 😉

File: src/library/Core.nim [hunk 2]
```diff
@@ -238,6 +238,17 @@ proc defineSymbols*() =
         attrs       = NoAttrs,
         returns     = {Any},
         example     = """
+            a: null
```
Comment (drkameleon) [inline src/library/Core.nim:241]: I think there is a problem with the `coalesce` documentation.

Basically, it's nothing but the [nil-coalescing operator](https://www.hackingwithswift.com/example-code/language/what-is-the-nil-coalescing-operator)...

Let me explain...

Let's say we have this:

```
x: attr "myAttr"  ; gets attribute `.myAttr` if it doesn't exist it returns null
if? not? null? x -> print x
else -> print "attr not found"
```

The equivalent with `coalesce` would be:

```
print coalesce attr "myAttr" "attr not found"
```

or... with `??`:

```
print (attr "myAttr") ?? "attr not found"
```

😉

**Comment by RickBarretto** [comment]
Sorry, I didn't get any Github's notification about these reviews before...
I'm just reading this now! 😵‍💫

**Review by drkameleon — COMMENTED** [review]
Awesome job! Let's move on with it! 🚀

**Comment by drkameleon** [comment]
> Sorry, I didn't get any Github's notification about these reviews before... I'm just reading this now! 😵‍💫

No worries! Happens to me all the time! lol

**Review by drkameleon — APPROVED** [review]

**Comment by drkameleon** [comment]
And... ready to merge! 🚀

**Pull request merged**